### ISSUE TITLE
Oculars Plug-in: Be able to define custom FOV (telrad) circles.

### ISSUE THREAD

**Issue by dimitri-rebrikov** [Stellarium/stellarium#1057]
**Current functionality** 
1. Activate the Oculars plug-in (if not already activated)
2. Press "Ctrl+B" or the "Telrad" button on the UI
3. The 3 Telrad FOV (Field of View) circles appear

**Improvement proposal (short/technical version)**
Extend the current functionality to be able to define different amount and degrees of the FOV circles of the Telrad simulation. 

**Improvement proposal (long version)**
The FOVs of the current circles are hard coded to 0.5, 2, 4 degrees making them only usable if you have the real Telrad piece. Making the degree and the amount of the circles configurable would make the functionality also usable for the manual object search using the telescope hardware configuration consisting of a finder scope and several oculars.  So the user will be able to know how to place the stars in the finder scope (or in the less magnified ocular) to get the desired object in the view of the (more magnified) ocular. 
The similar functionality is already implemented in the Telescope Control plug-in (see https://www.cloudynights.com/topic/152847-simulating-telescope-in-stellarium/#entry3490499) but it is not convenient for the “normal” manual search as the circles do not stay in the middle of the screen and you always need to call the Telescope Control dialog to slew the circles to the new position.

**Comment by dimitri-rebrikov** [comment]
I already spied in the source code. The change seems to be relative simple so even I as a Java coder could try to accomplish this. I could even implement the first variant without the extension of the configuration UI (i.e. just edit the ini file). 

So my question to the core team is: is this change request reasonable?

**Comment by gzotti** [comment]
The thing is called "Telrad" for historical reasons. It should be easy to replace/enhance this by a configurable 4-element Vec4f. Only non-zero elements are drawn. Default should obviously be 0.5,2,4,0. There is enough space in the config GUI for 4 spinboxes to compose this vector. And don't forget to describe this in the User Guide :-)

**Comment by alex-w** [comment]
The Telrad feature are simulated the Telrad device and adding of  support of  "strange" Telrad's in not very good idea IMHO.

I really do not understand target of this proposal.

**Comment by dimitri-rebrikov** [comment]
Hi Alex, my target is to provide more convenience for the people trying to find low magnitude objects using an alt-azimuth mount without any positioning support (i.e. without Go-to, Push-to etc.). 

An example how I searched for the 2 current comets C/2017 T2 and C/2019 Y4:
1. At the first I used the finder scope (FOV: 7.5°) and the Starhopping for the raw positioning. 
2. Then I used the 25mm ocular (FOV: 1.43°) for more precise positioning as I could see more starts through it. 
3. Finally I used the 10mm ocular (FOV: 0.57°) to make the final positioning along the stars I could not see with the 25mm and then tried to recognize the comet.

During these steps I was relying to the Stellarium sky map in the corresponding FOV/Magnification to align the star positions I see trough the optic with the star positions I see in the map. 

Be able to see the FOV circles (in my case 7.5°, 1.43°, 0.57°) in the Stellarium would make the search faster and more convenient. Using the ocular simualtion and turning the semi-transparent mask on provides a kind of this functionality but not as convenient as it would be if I just have all the 3 FOV circles displayed in the middle of the sky map. Just like the Telrad simulation does. 

So my idea was to extend the current Telrad functionality to be able to switch from the default FOV circles (0.5, 2.0, 4.0) to the custom define some. 

I understand that one might see this change as a misuse of the Telrad function. But IMHO this is the way to get the desired functionality with the least effort. And the users which just like to use the standard Telrad will be not disturbed because of the default settings. I would also add the button "set to default" to restore the default values once changed.

The "golden bullet" would be to be able to define/select the optic configurations one like to use for the sky session (i.e. finder, lens/barlow/telescope combinations) and then just turn the FOV simulations for all of them. But this is IMHO a lot of implementation effort and even a lot of re-engineering in the plug-ins UI and code or even a reason for a new separate plug-in.

**Comment by gzotti** [comment]
That's indeed a funny abuse. However, there are also other zero-magnification devices which could be configured with this approach. So, together with a "Standard Telrad" resetting button, I see no problem to proceed as described. 
A new plugin is not reasonable. Additions should go into the existing, with "reasonable" modifications. There are some technical challenges to solve before greater changes should be invented, though. (Get rid of the deprecated QSignalMapper, anybody?)

**Comment by alex-w** [comment]
@gzotti could you please check tooltips for new tool?

**Comment by gzotti** [comment]
Yes, I am extending also to 4 rings...

**Comment by alex-w** [comment]
Georg, now Telrad may have only one set of FOV. If you want to define some specific FOV set it's OK, but after switching to default Telrad FOV values you cannot quick restore custom values.

**Comment by dimitri-rebrikov** [comment]
I tested both solutions and found both of them sufficient for me.
I can imagine to use the 4 FOVs in a session (i.e one Finder and 3 oculars).
I have no experience with Telrads but I can imagine if someone has a Telrad, he/she uses it for the raw positioning and the ocular FOVs could be still quite useful for the fine alignment. But then the switch Telrad/Ocular FOV needs to be placed to the main screen just like the current switch between the Ocular and the Telrad simulation, which is IMHO a bigger effort to implement the right way as it touches the basic concepts of the Ocular plug-in. 
Anyway I impressed by your speed and already have the current master version compiled and copied to my laptop. Thank you guys!
